checker/src/main/java/org/checkerframework/checker/signature/qual/GenericSignature.java

```diff
@@ -0,0 +1,31 @@
+package org.checkerframework.checker.signature.qual;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * Represents a generic type signature as defined in <a
+ * href=https://docs.oracle.com/javase/specs/jvms/se9/html/jvms-4.html#jvms-4.7.9.1>JVM specs</a>.
+ * Can be one of the following:
+ *
+ * <p>1. JavaTypeSignature
```

Comment:
Here, I think it will lead to less documentation duplication to move the documentation to the subtypes, and link from here to the subtypes (there aren't currently links).